torchci/clickhouse_queries/lf_rollover_health/query.sql

```diff
@@ -1,144 +1,96 @@
--- !!! Query is not converted to CH syntax yet.  Delete this line when it gets converted
-
-WITH normalized_jobs AS (
+-- This query is used to generate a chart on HUD to compare the success, cancellation, and duration rates
+-- of jobs in the Meta fleet vs jobs in the LF fleet over time.
+WITH
+normalized_jobs AS (
   SELECT
     j.started_at,
-    ROUND(DATE_DIFF('MINUTE', PARSE_TIMESTAMP_ISO8601(j.started_at), PARSE_TIMESTAMP_ISO8601(j.completed_at)), 1) as duration_min,
-    if(
-        strpos(l.label, 'amz2023.') = 0,
-        l.label,
-        CONCAT(
-            substr(l.label, 1, strpos(l.label, 'amz2023.') - 1),
-            substr(l.label, length('amz2023.') + strpos(l.label, 'amz2023.'))
-        )
-    ) as label,
-    REGEXP_EXTRACT(j.name, '([^,]*),?', 1) as job_name, -- remove shard number and label from job names
+    ROUND(dateDiff('minute', j.started_at, j.completed_at), 1) as duration_min,
+    l as label,
+    extract(j.name, '[^,]*') as job_name, -- remove shard number and label from job names
     j.workflow_name,
     j.conclusion,
-      DATE_TRUNC(:granularity, PARSE_TIMESTAMP_ISO8601(j.started_at))
-    AS bucket,
+    toStartOfInterval(j.started_at, INTERVAL 1 DAY) AS bucket
   FROM
-      commons.workflow_job j
-      CROSS JOIN UNNEST(j.labels as label) as l
-  WHERE 1=1
-    AND j.labels is not NULL
-    AND j._event_time > CURRENT_DATETIME() - DAYS(:days_ago)
+    workflow_job as j
```

Comment:
final?

Reply:
doh, sorry

Reply:
Based on offline conversation, if you are willing to eat the cost of not using final (maybe some incorrect rows, dup rows, esp for recent data), then you don't need final.  A comment about not needing it would be nice tho

Reply:
As discussed offline, we'll leave this query without FINAL.  Added comments to the query explaining the logic